.github/workflows/release-assets.yml

```diff
@@ -0,0 +1,286 @@
+name: Release assets
+
+on:
+  create:
+    tags: ['v*']
+    
+  release:
+    types: [created]
+
+jobs:
+
+  ########
+  # info #
+  ########
+
+  info:
+    name: Print information
+    runs-on: ubuntu-latest
+    steps:
+    - name: Print context
+      run: echo Packaging for release ${{ github.ref }}
+    - name: Print event
+      run: echo Triggered by ${{ github.event_name }}
+
+  ##################
+  # create release #
+  ##################
+
+  create_release:
+    name: Create release from tag
+    runs-on: ubuntu-latest
+    needs: info
+    steps:
+    - name: Create release (optional)
+      if: github.event_name == 'create' # tag created, promote to release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.ref }}
+        release_name: ${{ github.ref }}
+        draft: false     # Must not be to be retrievable later
+        prerelease: true # Require manual validation
+
+  #########
+  # Linux #
+  #########
+
+  linux:
+    name: Prepare for ${{ matrix.tag }}
+    runs-on: ubuntu-latest
+    needs: create_release
+
+    strategy:
+      matrix:
+        tag: # Those are our dockerhub alire/gnat:tag machines
+            - centos-latest-community-latest
+            - debian-stable
+            - ubuntu-lts
+            # - community-latest # Not really a target platform
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v1
+
+    - name: Check out submodules
+      run: git submodule update --init --recursive
+
+    - name: Pull docker image
+      run: docker pull alire/gnat:${{ matrix.tag }}
+
+    - name: Build
+      run: >
+        docker run -v${PWD}:/alire -w /alire
```

Comment:
I don't like that alr is rebuilt just in this workflow. The binary released is not that same that the one tested.

Can't we have this publishing actions optional in the usual workflow scripts?

Reply:
With current workflow limitations it will be doable but ugly. I'll try to come with a decent solution.

Reply:
In that case I prefer a manual upload of the release binaries from the CI runs.

Reply:
What about testing these executables again? That would add only a step per job.

Reply:
> With current workflow limitations it will be doable but ugly.

Actually I'm not sure it's doable, as PR artifacts are totally disconnected from commits/releases, and the commit changes after the merge.

Reply:
> That would add only a step per job.

Actually a few, to set up the virtualenv